lib/git.js

```diff
@@ -60,13 +60,25 @@ exports.getCommits = function (options) {
 
     var parsed = commit.subject.match(COMMIT_PATTERN);
 
-    if (!parsed || !parsed[1] || !parsed[4]) {
+    if (!parsed || !parsed[1] || !parsed[3]) {
       return null;
     }
 
     commit.type = parsed[1].toLowerCase();
-    commit.category = parsed[3] || '';
-    commit.subject = parsed[4];
+    commit.category = parsed[2] || '';
+    commit.subject = parsed[3];
+
+    if (parsed[4]) {
+      parsed[4].toLowerCase().split(',').forEach(function (flag) {
+        flag = flag.trim();
+
+        switch (flag) {
```

Comment:
Thoughts on just making this `flag.trim()` instead of reassigning `flag`?

Reply:
The trimmed version needs to be captured either via reassignment or creating a new variable since it used in more than one place (line 77 assigns the value to `type` in the case of a breaking flag). As far as reassignment goes, in this case I don't see as an issue in the same way reassigning a function parameter for a module API would be for two reasons:

1. This is essentially a pipeline that is parsing and formatting the flags prior to processing the logic for each flag. I could refactor the trim to occur in a `map` prior to the switch but I didn't see the need to loop the split flag array twice at this time.
2. Since `flag` is a primitive value, reassigning it does not have any side effects (as modifying an object property would).

Reply:
The trimmed version needs to be captured either via reassignment or creating a new variable since it used in more than one place (line 77 assigns the value to `type` in the case of a breaking flag). As far as reassignment goes, in this case I don't see as an issue in the same way reassigning a function parameter for a module API would be for two reasons:

1. This is essentially a pipeline that is parsing and formatting the flags prior to processing the logic for each flag. I could refactor the trim to occur in a `map` prior to the switch but I didn't see the need to loop the split flag array twice at this time.
2. Since `flag` is a primitive value, reassigning it does not have any side effects (as modifying an object property would).